File: aoc-day3-p1.js
// Put csv input here:
let input = '';
// Stop numbers from being at ends of lines
input = input.replaceAll(',', '.,');
input = input.split(',');
let out = 0;
input.forEach((line, indexL)=>{
	let split = line.split("");
  let constructedNumber = '';
  let valid = false;
  split.forEach((sChar, indexC)=>{
  	if (isNaN(parseInt(sChar))){
    	out += valid ? parseInt(constructedNumber) : 0;
      valid = false;
      constructedNumber = '';
    }
    else {
    	constructedNumber += sChar;
      let negL = indexL - 1 >= 0;
      let negC = indexC - 1 >= 0;
      let posL = indexL + 1 < input.length;
      let posC = indexC + 1 < split.length;
      let adjacencyArr = [... negL ? [... negC ? [input[indexL-1][indexC-1]] : [], input[indexL-1][indexC], ... posC ? [input[indexL-1][indexC+1]] : []] : [],
      										... negC ? [input[indexL][indexC-1]] : [], ... posC ? [input[indexL][indexC+1]] : [],
                          ... posL ? [... negC ? [input[indexL+1][indexC-1]] : [], input[indexL+1][indexC], ... posC ? [input[indexL+1][indexC+1]] : []] : []];
      if (!valid){
      	adjacencyArr.forEach((adjChar)=>{
          valid = adjChar !== '.' && isNaN(parseInt(adjChar)) ? true : valid;
        });
      }
    }
  });
});
console.log(out);

File: 2021/aoc-day3-p1.py
# Put csv input here:
x = ""

y = x.split(",")
hold = [0,0,0,0,0,0,0,0,0,0,0,0]
recs = len(y)
for z in y:
    for a in range(12):
        hold[a] += int(z[a])
gamma = ""
epsilon = ""
for b in hold:
    if b > recs/2:
        gamma += "1"
        epsilon += "0"
    else:
        gamma += "0"
        epsilon += "1"
print("Result: " + str(int(gamma, 2) * int(epsilon, 2)))
